### PR TITLE
Convoy and garrison rework, part 2

### PR DESCRIPTION
**Pull request by wurzel0701**
The part of the convoy and garrison rework, that should be in 2.2. Will simulate reinforcements convoy across the whole map, dynamic spawn in and out and warlevel adapted unit selection.

Difference between this and part 1 are the following point:
- [DONE] Improvement of the cyclic spawn system
- [DONE] Integration into the convoy system, to use reinforce convoys
- [DONE] Improve the convoy system to spawn convoys
- [DONE] Save and load mechanics
- [DONE] Add mechanic on convoy arrival
- [DONE] On marker change mechanic
- [DONE] Write the "AI", which replenishes airports
- [DONE] ~~Rework AIVehInit to work for convoy vehicles~~
- [DONE] Add simple fight mechanic for roadblock engagements

### CODE REVIEW

**Comment by wurzel0701** [comment]
Code finished, needs alot of testing I assume

File: A3-Antistasi/functions/Convoy/fn_convoyDebug.sqf [hunk 1]
```diff
@@ -7,7 +7,7 @@
 
 if(isDedicated) exitWith {};
 
-if(isMultiplayer && {!(serverCommandAvailable "#logout")}) exitWith {hint "Only server admins can execute the convoy debug!"};
+//if(isMultiplayer && {!(serverCommandAvailable "#logout")}) exitWith {hint "Only server admins can execute the convoy debug!"};
```
[the remark this answers is not in the document]
Comment (Spoffy) [reply A3-Antistasi/functions/Convoy/fn_convoyDebug.sqf]: Closed, as clearly you need debug console access to run it! :)

**Comment by Spoffy** [comment]
Merged as of b0c3cc2bf3154c5bef651d935b8074bffff9747a

**Comment by Spoffy** [comment]
Still has issues:
- Nowhere near enough recruitment at airfields to meet demand
- Attempting to spawn land convoys for targets over oceans

File: A3-Antistasi/functions/CREATE/fn_createAISite.sqf [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+_patrolMarker = createMarkerLocal [format ["%1_patrol_%2", _marker, random 100], _markerPos];
+_patrolMarker setMarkerShapeLocal "ELLIPSE";
+_patrolMarker setMarkerSizeLocal _patrolMarkerSize;
+_patrolMarker setMarkerTypeLocal "hd_warning";
+_patrolMarker setMarkerColorLocal "ColorRed";
+_patrolMarker setMarkerBrushLocal "DiagGrid";
+_patrolMarker setMarkerDirLocaL (markerDir _marker);
+
+if(!debug) then
```
[the remark this answers is not in the document]
Comment (wurzel0701) [reply A3-Antistasi/functions/CREATE/fn_createAISite.sqf]: Only local, but I will fix it
Should be no problem on servers

File: A3-Antistasi/functions/CREATE/fn_milBuildings.sqf [hunk 3]
```diff
@@ -15,6 +15,19 @@ _soldiers = [];
 _groupX = createGroup _sideX;
 _typeUnit = if (_sideX==Occupants) then {staticCrewOccupants} else {staticCrewInvaders};
 
+//New system to place helis, does not care about heli types currently
+private _helicopterTypes = [];
```
[the remark this answers is not in the document]
Comment (wurzel0701) [reply A3-Antistasi/functions/CREATE/fn_milBuildings.sqf]: Will add it to the todo list

File: A3-Antistasi/functions/Garrison/fn_shouldReinforce.sqf [hunk 4]
```diff
@@ -0,0 +1,31 @@
+params ["_base", "_target"];
+
+private ["_isAirport", "_side", "_targetIsBase", "_reinfMarker", "_targetReinforcements", "_reinfCount", "_maxSend"];
+
+_isAirport = _base in airportsX;
+_side = sidesX getVariable [_base, sideUnknown];
+
+//To far away for land convoy or not the same island
+if(!_isAirport && {(getMarkerPos _base) distance2D (getMarkerPos _target) > distanceForLandAttack || {!([_base, _target] call A3A_fnc_isTheSameIsland)}}) exitWith {false};
```
[the remark this answers is not in the document]
Comment (wurzel0701) [reply A3-Antistasi/functions/Garrison/fn_shouldReinforce.sqf]: if (isNotAirport and cannotSendLand)

Bases can't send air convoy currently

**Comment by wurzel0701** [comment]
The recuitment limit even 50% higher than the original (12 to 8) 
And bases can recruit too (8 instead of 0)
That should be enough

**Comment by wurzel0701** [comment]
Is it still sending land convoys over sea?

**Pull request merged**